File: frontend/src/types/flight.ts
export type RiskLevel = 'LOW' | 'MEDIUM' | 'HIGH' | 'CRITICAL';

export interface Airport {
  id: string;
  code: string;
  name: string;
  city: string;
  country: string;
  timezone: string;
}

export interface Airline {
  id: string;
  code: string;
  name: string;
  country?: string;
}

export interface Aircraft {
  id: string;
  tail_number: string;
  model: string;
  total_capacity: number;
}

export interface DelayPrediction {
  id: string;
  delay_probability: number;
  predicted_delay_minutes: number;
  risk_level: RiskLevel;
  contributing_factors: string[];
  model_version: string;
  created_at: string;
}

export interface Flight {
  id: string;
  flight_number: string;
  airline: Airline;
  aircraft: Aircraft;
  origin: Airport;
  destination: Airport;
  scheduled_departure: string;
  scheduled_arrival: string;
  actual_departure?: string;
  actual_arrival?: string;
  duration_minutes: number;
  status: string;
  base_price: number;
  available_seats: number;
  delay_prediction?: DelayPrediction;
}

export interface FlightSearchParams {
  origin?: string;
  destination?: string;
  departure_date?: string;
  airline_code?: string;
  min_price?: number;
  max_price?: number;
  risk_level?: string;
  sort_by?: string;
  page?: number;
  limit?: number;
}

export interface PaginatedFlightResponse {
  items: Flight[];
  total: number;
  page: number;
  limit: number;
  total_pages: number;
}

File: backend/app/schemas/flight.py
from datetime import datetime
from typing import List, Optional
from pydantic import BaseModel, ConfigDict


class AirportResponse(BaseModel):
    id: str
    code: str
    name: str
    city: str
    country: str
    timezone: str

    model_config = ConfigDict(from_attributes=True)


class AirlineResponse(BaseModel):
    id: str
    code: str
    name: str
    country: Optional[str] = None

    model_config = ConfigDict(from_attributes=True)


class AircraftResponse(BaseModel):
    id: str
    tail_number: str
    model: str
    total_capacity: int

    model_config = ConfigDict(from_attributes=True)


class PredictionResponse(BaseModel):
    id: str
    delay_probability: float
    predicted_delay_minutes: int
    risk_level: str
    contributing_factors: List[str] = []
    model_version: str
    created_at: datetime

    model_config = ConfigDict(from_attributes=True)


class FlightResponse(BaseModel):
    id: str
    flight_number: str
    airline: AirlineResponse
    aircraft: AircraftResponse
    origin: AirportResponse
    destination: AirportResponse
    scheduled_departure: datetime
    scheduled_arrival: datetime
    actual_departure: Optional[datetime] = None
    actual_arrival: Optional[datetime] = None
    duration_minutes: int
    status: str
    base_price: float
    available_seats: int
    delay_prediction: Optional[PredictionResponse] = None

    model_config = ConfigDict(from_attributes=True)


class PaginatedFlightResponse(BaseModel):
    items: List[FlightResponse]
    total: int
    page: int
    limit: int
    total_pages: int


class LiveSyncResponse(BaseModel):
    success: bool
    source: str
    message: str
    total_synced: int
    created_count: int
    updated_count: int
    flights: List[FlightResponse] = []
